htdocs/multicurrency/multicurrency_rate.php

```diff
@@ -111,46 +118,46 @@
 		$currencyRate_static->rate = $rateinput;
 
 		$result = $currencyRate_static->create(intval($fk_currency));
-		if ($result > 0) {
-			setEventMessages($langs->trans('successRateCreate', $multicurrency_code), null);
+		if ($result) {
+			setEventMessage($langs->trans('successRateCreate', $multicurrency_code));
 		} else {
 			dol_syslog("currencyRate:createRate", LOG_WARNING);
-			setEventMessages($currencyRate_static->error, $currencyRate_static->errors, 'errors');
+            setEventMessages($currencyRate_static->error, $currencyRate_static->errors, 'errors');
 		}
 	} else {
-		setEventMessages($langs->trans('NoEmptyRate'), null, "errors");
+		setEventMessage($langs->trans('NoEmptyRate'), "errors");
 	}
 }
 
-if ($action == 'update') {
+if ($actionModify){
 	$currencyRate = new CurrencyRate($db);
-	$result = $currencyRate->fetch($id_rate_selected);
-	if ($result > 0) {
-		$currency_static = new MultiCurrency($db);
+	$result = $currencyRate->fetch(GETPOST('rowid', 'alpha'));
+	if ( $result > 0){
+		$currency_static  = new MultiCurrency($db);
 		$fk_currency = $currency_static->getIdFromCode($db, $multicurrency_code);
-		$currencyRate->date_sync = $dateinput;
+		$currencyRate->date_sync = $db->escape(GETPOST('dateinput', 'alpha'));
```

Comment:
date_sync is a properyt in memory so it if is a date it must contains a timestamp so a value that is result of dol_mktime or a result of db->jdate.
Old code seems better.

Reply:
old rereplaced